app/livechat/server/lib/Livechat.js

```diff
@@ -1097,6 +1097,25 @@ export const Livechat = {
 		return true;
 	},
 
+	getRoomMessages({ rid }) {
+		check(rid, String);
+
+		const isLivechat = Promise.await(Rooms.findByTypeInIds('l', [rid])).count();
+
+		if (!isLivechat) {
+			return [];
+		}
+
+		const ignoredMessageTypes = ['livechat_navigation_history', 'livechat_transcript_history', 'command', 'livechat-close', 'livechat-started', 'livechat_video_call'];
+		const messageCursor = Promise.await(Messages.findVisibleByRoomIdNotContainingTypes(rid, ignoredMessageTypes, { sort: { ts: 1 } }));
```

Comment:
```suggestion
		const messageCursor = Promise.await(Messages.findVisibleByRoomIdNotContainingTypes(rid, ignoredMessageTypes, { sort: { ts: 1 } })).toArray();
```

Reply:
I think we can convert a cursor to array like this. This way we don't have to loop through the cursor and manually add each message to an empty array

Reply:
The thing is this method call returns a cursor that does not have a `toArray` method included, unfortunately. That's why I'm looping through the query's contents.

The error:
```
TypeError: cursor.toArray is not a function                                                                                                                                                                                               
     at Object.getRoomMessages (app/livechat/server/lib/Livechat.js:1055:27)
```

Reply:
I believe you don't need to call toArray here. Would be enough to call `.fetch` on the returned cursor, that will return an array

Reply:
Didn't know about fetch. I'll try it